### PR TITLE
WT-3669 Check for changes rolled back during reconciliation.

### PR DESCRIPTION
**Pull request by michaelcahill**

### CODE REVIEW

**Comment by michaelcahill** [comment]
@keithbostic, I haven't seen these assertions fire, but I think the change to check `uncommitted` during reconciliation is necessary.  Check the ticket (and WT-3406) for details.  I don't really need a review: I mostly want to leave tests running against this branch overnight.

File: src/btree/bt_split.c [hunk 1]
```diff
@@ -1665,6 +1665,9 @@ __wt_multi_to_ref(WT_SESSION_IMPL *session,
 	 * overrides WT_REF_DISK.
 	 */
 	if (multi->las_pageid != 0) {
+		WT_ASSERT(session, multi->disk_image == NULL);
+		WT_ASSERT(session, multi->addr.addr != NULL);
```
Comment (keithbostic) [inline src/btree/bt_split.c]: @michaelcahill, the assert of `addr.addr` backs out the change in WT-3663 (#3739)?

I definitely saw pages that were only lookaside records, without an image or backing block.

Comment (michaelcahill) [reply src/btree/bt_split.c]: Thanks @keithbostic, that was just sanity checking, I didn't intend for it to be merged.  Interestingly, I haven't seen it fire yet.  I've backed it out regardless.

**Comment by agorrod** [comment]
lgtm with one nit about wanting a comment explaining the first assertion.

**Comment by michaelcahill** [comment]
autobuild: retest this please

**Comment by agorrod** [comment]
Now that tests have passed, merge when you are ready

**Comment by michaelcahill** [comment]
Thanks @agorrod, merging.

**Pull request merged**